src/time/YamlDateTime.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_TIME_YAMLDATETIME_H
+#define SQUID_SRC_TIME_YAMLDATETIME_H
+
+#include <iosfwd>
+#include <sys/time.h>
+
+namespace Time {
+
+/// Output onto an ostream a yaml-formatted datetime string (UTC)
+///  see https://yaml.org/type/timestamp.html
+class YamlDateTime
```

Comment:
Do we really need to put YamlDateTime inside Time namespace? The class name already has "Time". And Yaml. And Date. Why pick Time out of those three? The namespace itself is highly questionable. I would keep this class outside that namespace.